src/main/java/qna/domain/Question.java

```diff
@@ -44,6 +49,22 @@ public boolean isOwner(User writer) {
         return this.writer.equals(writer);
     }
 
+    public void checkOwner(User writer) throws CannotDeleteException {
+        if (!isOwner(writer)) {
+            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
+        }
+    }
+
+    public List<DeleteHistory> delete(Answers answers, User writer) throws CannotDeleteException {
```

Comment:
사소하지만 일급 컬렉션을 반환해보는 건 어떨까요 ?
```suggestion
    public DeleteHistories delete(Answers answers, User writer) throws CannotDeleteException {
```